src/Kiota.Builder/Writers/CSharp/CSharpConventionService.cs

```diff
@@ -231,7 +243,7 @@ public bool IsPrimitiveType(string typeName)
     }
     public override string GetParameterSignature(CodeParameter parameter, CodeElement targetElement, LanguageWriter? writer = null)
     {
-        var parameterType = GetTypeString(parameter.Type, targetElement);
+        var parameterType = GetTypeString(parameter.Type, targetElement, true, false);
```

Comment:
Passing `false` will cause regression for value types (int,bool etc) and enums which have the nullable marker by default. See the `NullableTypes` collection at the top of the file. In such a scenario we need not add the maker as its already there. 
Omiting the information for these types leads to methods generated with the missing marker and leading to compile errors.

Reply:
I think that's a bug with the inclusion of .NET versions that don't support nullable types. That code won't compile anyway, right?

Reply:
Can you help me create a test case and add exclusions until we can fix it permanently? Also, I saw that the `GetParameterSignature` only has 3 uses in the C# language. I've updated all those uses.

Reply:
I've made updates and added unit tests for the expectations of the code specified.